### ISSUE TITLE
How to get cookies in the websocket initiation handler

### ISSUE THREAD

**Issue by robert-stuttaford** [sunng87/ring-jetty9-adapter#43]
Adapted from the websocket example in this repo:

```clojure
(ns rj9a.websocket
  (:gen-class)
  (:require [ring.adapter.jetty9 :as jetty]))

(defn dummy-app [req] {:body "<h1>It works</h1>" :status 200})

(defn websocket-accept [req]
  ;; 1. how do we get `:cookies` to be in this `req`?
  {:on-text
   (fn [ws text]
     ;; 2. you can call `(req-of ws)` here to get the upgraded request
     (jetty/send! ws text))})

(defn -main [& args]
  (jetty/run-jetty dummy-app
                   {:port       5000
                    :websockets {"/path" websocket-accept}}))
```

How do we get `:cookies` to be available on `req` inside `websocket-accept` as noted at 1. in this example?

My use-case is that we'll only start the websocket if the session is authenticated, which we need to inspect the cookies to be able to verify.

**Comment by sunng87** [comment]
@robert-stuttaford have you ever printed `req` in `websocket-accept`? Because middlewares are not available in this handler, you may not be able to use `:cookies` on req. But it should be available as a header. In my local `wscat` test, the `req` is:

```edn
{:uri /path3/, :query-string nil, :origin nil, :host localhost, :request-method :get, :headers {:connection Upgrade, :host localhost:5000, :sec-websocket-extensions permessage-deflate; client_max_window_bits, :sec-websocket-key hGCpMmwoMbKRrcxZYs2q6Q==, :sec-websocket-version 13, :upgrade websocket}}
```

**Comment by robert-stuttaford** [comment]
Goodness, you're right. I'll see if I can get the right ring cookie middleware functions to run manually. Thanks!

**Comment by sunng87** [comment]
By the way, it seems I had a mistake in the `headers` where names should be string instead of keyword. This issue will stop you using cookie middlewares. Let me create a fix tomorrow for that.

**Comment by robert-stuttaford** [comment]
I just came back to report this is why I'm not seeing cookies, haha. Yes please :-)

**Comment by robert-stuttaford** [comment]
Here you go: https://github.com/sunng87/ring-jetty9-adapter/pull/44

**Comment by sunng87** [comment]
Fixed in #44, released in 0.12.8
